### PR TITLE
Test: Always provided optional module input parms in end2end tests

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

File: plugins/modules/zhmc_adapter_list.py [hunk 1]
```diff
@@ -156,8 +156,8 @@
   _faked_session:
     description:
       - "An internal parameter used for testing the module."
-    required: false
```
Comment (EdGue42) [inline plugins/modules/zhmc_adapter_list.py:159]: why reordering?

File: plugins/modules/zhmc_crypto_attachment.py [hunk 2]
```diff
@@ -190,8 +190,8 @@
   _faked_session:
     description:
       - "An internal parameter used for testing the module."
-    required: false
```
Comment (EdGue42) [inline plugins/modules/zhmc_crypto_attachment.py:193]: same: this change doesn't look necessary?

File: plugins/modules/zhmc_nic_list.py [hunk 3]
```diff
@@ -117,8 +117,8 @@
   _faked_session:
     description:
       - "An internal parameter used for testing the module."
-    required: false
     type: raw
+    required: false
```
Comment (EdGue42) [inline plugins/modules/zhmc_nic_list.py:121]: same same

File: plugins/modules/zhmc_adapter_list.py (hunk 1 above)
Comment (andy-maier) [reply plugins/modules/zhmc_adapter_list.py:159]: It makes it easier for me to check that all optional parms have a default value specified, if default is directly below required.

File: plugins/modules/zhmc_crypto_attachment.py (hunk 2 above)
Comment (andy-maier) [reply plugins/modules/zhmc_crypto_attachment.py:193]: see previous comment for the reason.

File: plugins/modules/zhmc_nic_list.py (hunk 3 above)
Comment (andy-maier) [reply plugins/modules/zhmc_nic_list.py:121]: see other comments

**Pull request merged**